### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by gnawf**
Surrounds all normal execution with try catch instead of just the query parsing (since we now have instrumentation that can "fail").

### CODE REVIEW

File: test/src/test/kotlin/graphql/nadel/tests/hooks/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    var instrumentationParams: NadelInstrumentationExecuteOperationParameters? = null
+    var resultBeforeFinalInstrumentation: ExecutionResult? = null
+
+    override fun makeNadel(engineType: NadelEngineType, builder: Nadel.Builder): Nadel.Builder {
+        return builder
+            .instrumentation(object : NadelInstrumentation {
+                override fun createState(parameters: NadelInstrumentationCreateStateParameters): InstrumentationState {
+                    return object : InstrumentationState {
+                        override fun toString(): String {
+                            return "so-annoying"
```
Comment (felipe-gdr) [inline test/src/test/kotlin/graphql/nadel/tests/hooks/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.kt:41]: hm, why is this annoying?

File: test/src/test/resources/fixtures/instrumentation/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        hello: World
+      }
+      type World {
+        id: ID
+        name: String
+      }
+      type Mutation {
+        hello: String
+      }
+      type Subscription {
```
Comment (felipe-gdr) [inline test/src/test/resources/fixtures/instrumentation/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.yml:18]: this `Subscription` type doesn't seem to be relevant to test and can be a bit distracting.

**Review by felipe-gdr — DISMISSED** [review]

File: test/src/test/kotlin/graphql/nadel/tests/hooks/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.kt (hunk 1 above)
Comment (gnawf) [reply test/src/test/kotlin/graphql/nadel/tests/hooks/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.kt:41]: Ah it's a copy paste from very similar test, I think I had issues with assertions when I was writing the test.

File: test/src/test/resources/fixtures/instrumentation/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.yml (hunk 2 above)
Comment (gnawf) [reply test/src/test/resources/fixtures/instrumentation/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.yml:18]: Ah yea, it's a copy paste from another test. I'll remove it.

**Review by felipe-gdr — APPROVED** [review]
